File: jobs-app-springboot/src/main/java/com/savaiz/jobs_app/entity/Subscription.java
package com.savaiz.jobs_app.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Table(name = "subscriptions")
@Data
public class Subscription {
    public Subscription() {}

    public Subscription(String userEmail, List<String> roles, List<String> regions) {
        this.userEmail = userEmail;
        this.roles = roles;
        this.regions = regions;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "user_email")
    private String userEmail;

    @ElementCollection
    @CollectionTable(name = "subscription_roles", joinColumns = @JoinColumn(name = "subscription_id"))
    @Column(name = "roles")
    private List<String> roles;

    @ElementCollection
    @CollectionTable(name = "subscription_regions", joinColumns = @JoinColumn(name = "subscription_id"))
    @Column(name = "regions")
    private List<String> regions;
}

File: jobs-app-python/models/subscription.py
from datetime import datetime


class Subscription:
    def __init__(self, user_email, roles, regions, created_at=None, updated_at=None):
        self.user_email = user_email
        self.roles = roles
        self.regions = regions
        self.created_at = created_at or datetime.now()
        self.updated_at = updated_at or datetime.now()

    def to_tuple(self):
        return (
            self.user_email,self.roles,self.regions
        )